README.md

```diff
@@ -1,82 +1,163 @@
+
+```swift 
+let numberOfStudents = 25.0
+let numberOfTeachers = Double(3)
+let classroomsNeeded = numberOfStudents / numberOfTeachers // 8.3333333333333339
+```
+
+Finally, we could have also solved the initial problem by making use of *type inference* and directly specify the result type we are expecting from our division:
+
+```swift
+let classroomsNeeded: Double = 25 / 3 // 8.3333333333333339
+```
+
+However, type inference is performed for each line individually, so it is not possible for the compiler to change the type of `numberOfStudents` or `numberOfTeachers` after the fact in order to fulfill an explicit type annotation on our `classroomsNeeded` result.
+
+Note: Floating-point types can represent a much wider range of values than integer types, and can store numbers that are much larger or smaller than can be stored in an `Int`. Because of how those numbers are stored internally, they don't have a fixed range of values, but rather a number of decimal digits they can represent with precision, in the case of `Double` this is at least 15 decimal digits.
+
+## Creating your own arithmetic functions
+
+You already learned about declaring functions and returning values from them in a previous lesson:
+
+```swift
+func twentyThree() -> Int {
+	return 23
+}
+
+twentyThree() // 23
+```
+
+Use this and your newly acquired knowledge about mathematical operators to create a function that adds two values.
+
+The function you have created should look somewhat like this:
+
+```swift
+func addNumber(first: Int, to second: Int) -> Int {
     return first + second
 }
+
+let result = addNumber(5, to: 3) // 8
 ```
-* I think this would be a good idea to introduce to them the `Double` type for the first time. I imagine a student messing around and trying to type something like `25 / 3` and seeing that its output is displaying as `8`.
-* Possibly have them type something like the following then asking them to option click the variable to see what its type is (where they will then see that it states `Double`).
+
+Note that we are giving the second parameter of the function two labels, an *external* one (`to`) and an *internal* one (`second`). This makes it possible to make the calling code more readable, almost feeling like a sentence, while also being able to easily distinguish the parameters inside the body of the function.
+
+The `return` statement is used to exit a function and return its value to the expression calling it. You can think of it as a way to extract a piece of code into its own environment and giving it a descriptive name, while hiding the details. Calling a function can be seen as inserting that piece of code back, so another way to express the code block above is:
+
 
 ```swift
-let funNumber = 25.0
+let result = 5 + 3 // 8
 ```
- 
+
+In this case, extracting the code into a function did not make much sense, because our code became neither more descriptive, nor did we make it shorter.
+
```

Comment:
Perfect. A student will also be asking why or making the assumption that when we show them something.. it's the way they should ALWAYS do it. This will help them make better decisions.